ldap/servers/plugins/sync/sync_persist.c

```diff
@@ -1072,6 +1073,11 @@ sync_send_results(void *arg)
     slapi_ch_free((void **)&strFilter);
     slapi_pblock_set(req->req_pblock, SLAPI_SEARCH_STRFILTER, NULL);
 
+    slapi_pblock_get(req->req_pblock, SLAPI_REQCONTROLS, ctrls);
+    if (ctrls) {
+        ldap_controls_free(ctrls);
```

Comment:
I think we should also set NULL there in the `if`:

```
slapi_pblock_set(req->req_pblock, SLAPI_REQCONTROLS, NULL);
```

Reply:
Hi @droideck 
The pblock is destroyed directly after this call so I felt there was no need to reset the request controls memory.

Reply:
Better to set to NULL just incase a new use arises where this assumption no longer holds true.